### PR TITLE
net, annotations-gen: Refactor GenerateFromSource

### PR DESCRIPTION
**Pull request by orelmisan**
### What this PR does
Return early in case the source pod does not have an ordinal naming scheme.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: pkg/network/pod/annotations/generator.go [hunk 1]
```diff
@@ -82,25 +82,23 @@ func (g Generator) Generate(vmi *v1.VirtualMachineInstance) (map[string]string,
 
 // GenerateFromSource generates ordinal pod interfaces naming scheme for a migration target in case the migration source pod uses it
 func (g Generator) GenerateFromSource(vmi *v1.VirtualMachineInstance, sourcePod *k8Scorev1.Pod) (map[string]string, error) {
-	annotations := map[string]string{}
+	if !namescheme.PodHasOrdinalInterfaceName2(multus.NetworkStatusesFromPod(sourcePod)) {
+		return nil, nil
```
Comment (ormergi) [inline pkg/network/pod/annotations/generator.go:86]: Is it safe returning nil instead of empty map?

Comment (orelmisan) [reply pkg/network/pod/annotations/generator.go:86]: The short answer is "yes".
This map is only used for read operations, so it is Ok to return `nil` here.
According to the Go specification:
https://go.dev/ref/spec#Map_types
> A nil map is equivalent to an empty map except that no elements may be added.

https://go.dev/ref/spec#Index_expressions
> For a of map type M:
> if the map is nil or does not contain such an entry, a[x] is the zero value for the element type of M

**Review by ormergi — APPROVED** [review]
LGTM

**Review by nirdothan — APPROVED** [review]
Thanks @orelmisan looks better than before although I expect it to function exactly the same

**Review by EdDev — COMMENTED** [review]
Nice.

/approve

**Pull request merged**